### PR TITLE
Minigame interface and Manager

### PR DESCRIPTION
**Pull request by justinrreyes**
## Description

Added a Minigame manager and Minigame Interface in order to control every minigame at the same location and allow minigame completion to affect other parts of the map.

## Type of Change

- [ ] Bug Fix
- [ ] New Feature
- [X] Enhancement
- [ ] Refactor
- [ ] Documentation
- [ ] Other (please specify):

## Related Issue

N/A

## Changes Made

- Created a MinigameManager script that will allow multiple minigames in the scene to interact together and affect the map.
- Created IMiniGame to interface with the manager allowing minigames to properly communicate with each other and the manager.

## Checklist

- [X] Code follows the project's coding style.
- [ ] If asset(s) added, they have been made into a prefab
- [X] Tests have been added/updated.
- [X] Documentation has been updated (if applicable).
- [X] No unused code or assets remain.
- [X] Unity project builds without errors.

## Screenshots (if applicable)

image

## Additional Notes

Currently the manager is only outputting a debug statement when completing the minigames, however there is a field that can be attached to any game object in the future to have the manager affect it.

### CODE REVIEW

**Review by EricPettie — APPROVED** [review]
## Pull Request Review

- **Functionality:**  
  - The generator minigame functioned well and correctly output the minigame win to console.

- **Resizing Scripts:**  
  - The minigame UI displays correctly.

- **Unused Code:**  
  -  None.

- **Code Style:**  
  - Aligns with the project.

- **Script Structure:**  
  - Everything is organized.

- **Documentation:**  
  - The comments are clear and describe the code well.

- **Build Integrity:**  
  - The game successfully built and ran.

---

### Summary
The generator minigame seems to be a great feature that functions well. It looks ready to merge.

**Comment by krain0** [comment]
I tried out your changes and everything appears to be functioning, the minigame is a good addition to the game. Once the merge conflict with sg1/Assets/Scenes/SampleScene.unity is resolved, I can leave a full review and we can merge it. The conflict with sg1/Assets/Scripts/GameLogic/PlayerMovement.cs won't be too hard to resolve manually. Now that everything is a prefab, you can just checkout the sample scene from main and commit that. We can reintegrate all of the new changes into the scene file in an upcoming PR.

**Review by uid106 — APPROVED** [review]
## Pull Request Review
- **Functionality**:
  - Adds a script for managing mini games and controlling when all tasks have been completed.

- **Unused Code**: 
  - Not yet attached to game objects, but will be in future pull requests.

- **Code Style**: 
   - Properly following project code style.

- **Script Structure**: 
   -  Script structure is well put together. 

- **Documentation**: 
   - Documentation is appropriate. 

- **Build Integrity**: 
  - Code compiles and runs with no issues, especially after fixing conflicts.

---

### Summary
The script seems to be of the correct style and very useful for future game objects, allowing it to be attached to objects for future mini games. It looks like the merge conflicts were fixed, so the PR should be good to merge, please check once more before merging.

**Pull request merged**
